service/raft/raft_group_registry.cc

```diff
@@ -335,6 +335,18 @@ static void ensure_aborted(raft_server_for_group& server_for_group, sstring reas
     }
 }
 
+future<> raft_group_registry::stop_server(raft::group_id gid, sstring reason) {
+    auto it = _servers.find(gid);
+
+    if (it == _servers.end()) {
+        throw raft_group_not_found(gid);
+    }
+
+    ensure_aborted(it->second, std::move(reason));
+    co_await std::move(*it->second.aborted);
```

Comment:
I think this could be called concurrently with `abort_server` which is called by `raft::server::configuration::on_background_error` for the group 0 server.

So the following could happen:
- here we move from the `aborted` future which causes it to turn into a ready future (I think? If not, and the behavior is undefined, it's even worse)
- while we're waiting, before we `erase`, concurrently `on_background_error` calls `abort_server` which sets `aborted` again
- then we `erase` the server, so this new `aborted` future is never waited for (it's discarded).

I think to prevent this, we should remove the server from the map before waiting for the future. Then `abort_server` won't find it.
But we cannot just `erase` it because that would destroy the server.

We should be able to move the server first, then erase, then wait for the abort to finish. Something like:
```cpp
auto srv = std::move(it->second);
_servers.erase(it);
co_await std::move(*srv.aborted);
```